feat/add put delete transaction endpoints

affan880 opened the pull request.

jackli921 committed:
Add POST /transactions endpoint

- added transaction related schema

- added related service and controller logic

- added related unit tests

- added Decimal.js to package.json to handle monetary data accurately

jackli921 committed:
Add POST /transactions endpoint

- added transaction related schema

- added related service and controller logic

- added related unit tests

- added Decimal.js to package.json to handle monetary data accurately

jackli921 committed:
Add POST /transactions endpoint

- added transaction related schema

- added related service and controller logic

- added related unit tests

- added Decimal.js to package.json to handle monetary data accurately

kayoMichael committed:
Add Docker pipeline and prod dockerfile

kayoMichael committed:
Add Docker pipeline and prod dockerfile

kayoMichael committed:
Change target branch

kayoMichael committed:
Change target branch

affan880 committed:
Add GET /transactions/:id endpoint

- Implemented getTransactions controller to fetch transactions by user ID
- Added getTransactionsSchema for input validation
- Updated transaction service to include getTransactions method
- Modified transaction routes to include the new GET endpoint

kayoMichael committed:
Refactor dockerfile and actions

Refactor Commit

kayoMichael committed:
Refactor dockerfile and actions

Refactor Commit

kayoMichael committed:
Refactor

kayoMichael committed:
Refactor syntax

kayoMichael committed:
Refactor GCS auth structure

kayoMichael committed:
Refactor server port

kayoMichael committed:
Refactor server port

kayoMichael committed:
Set client host for Cors

kayoMichael committed:
Set client host for Cors

kayoMichael committed:
Add conditions to job

kayoMichael committed:
Add conditions to job

kayoMichael committed:
Refactor syntax

kayoMichael committed:
Refactor Syntax error

jackli921 committed:
Add GET /categories Endpoint

- added categoryController, categoryService, categoryRouter

- added related unit tests

jackli921 committed:
Add requireAuth to protected API paths

- added requireAuth middleware to protected endpoints

- added mock requireAuth middleware to bypass it for controller unit tests

jackli921 committed:
Add requireAuth to protected API paths

- added requireAuth middleware to protected endpoints

- added mock requireAuth middleware to bypass it for controller unit tests

affan880 committed:
Remove unnecessary await and add sorting to getTransactions

jackli921 reviewed (commented):
I think you should rebase and clean up your commit history so that only one commit is needed to be added. 
It's hard to tell what changes applies to this PR specifically.

affan880 committed:
feat: Add PUT and DELETE endpoints for transactions - Add PUT /users/:id/transactions/:id endpoint to update transactions - Add DELETE /users/:id/transactions/:id endpoint to delete transactions - Add validation schemas for update and delete operations - Add service methods for update and delete operations - Add controller methods with proper error handling - Ensure users can only update/delete their own transactions - All endpoints require authentication via JWT cookies - Resolved merge conflicts with development branch

affan880 committed:
Merge branch 'development' into feat/add-put-delete-transaction-endpoints

affan880 committed:
fix: Resolve parsing errors in bun.lock and duplicate declaration in transaction.controller.ts

kayoMichael committed:
change workflow order

kayoMichael committed:
Refactor CORS config

kayoMichael committed:
Refactor deployment file

kayoMichael committed:
configure env variables

kayoMichael committed:
configure env variables

kayoMichael committed:
Add debug line

kayoMichael committed:
Temp Hard code

kayoMichael committed:
Temp Hard code

kayoMichael committed:
Add text to check for changes

kayoMichael committed:
Add text to check for changes

kayoMichael committed:
Refactor Client host

kayoMichael committed:
Refactor Client host

kayoMichael committed:
refactor cors

kayoMichael committed:
Configure CORS

kayoMichael committed:
Turn off CORS

kayoMichael committed:
Turn off CORS

kayoMichael committed:
Turn off CORS

kayoMichael committed:
Turn off CORS

kayoMichael committed:
Add logging

kayoMichael committed:
Add logging

kayoMichael committed:
Add trust proxy true

kayoMichael committed:
Add trust proxy true

kayoMichael committed:
Refactor user controller

kayoMichael committed:
Refactor user controller

kayoMichael committed:
Add console log

kayoMichael committed:
Add console log

kayoMichael committed:
Add ngin.conf

kayoMichael committed:
Refactor back test change

kayoMichael committed:
Refactor back test change

kayoMichael committed:
Add no cache

kayoMichael committed:
Eslint

kayoMichael committed:
Eslint

jackli921 committed:
Add Create Budget Endpoint

- added controller, service, and new router

- added related unit tests

jackli921 commented on apps/server/src/routes/transaction.routes.ts:
```diff
@@ -1,9 +1,17 @@
 transactionRouter.post('/', createTransaction);
+transactionRouter.get('/:id/transactions', getTransactions);
+transactionRouter.patch('/:userId/transactions/:id', updateTransaction);
+transactionRouter.delete('/:userId/transactions/:id', deleteTransaction);
```
i think for delete endpoint, we can just use /transaction/:id, the id will just point to the single record in the transaction table. We don't need to know which user it belongs to

jackli921 commented on apps/server/src/routes/transaction.routes.ts:
```diff
@@ -1,9 +1,17 @@
 
 transactionRouter.post('/', createTransaction);
+transactionRouter.get('/:id/transactions', getTransactions);
+transactionRouter.patch('/:userId/transactions/:id', updateTransaction);
```
we can just use `/transaction/:id` here as well and `id` just refers to the id of that specific transaction reord. We don't ned to know which user it belongs to.

jackli921 commented on apps/server/src/routes/transaction.routes.ts:
```diff
@@ -1,9 +1,17 @@
 const transactionRouter = Router();
 
 transactionRouter.post('/', createTransaction);
+transactionRouter.get('/:id/transactions', getTransactions);
```
if it's for getting all user transactions, it should be `transactions/user/:id` which i think you already implemented in somewhere else?

affan880 committed:
refactor: transaction update and delete endpoints by removing userId parameter

Shaimaa01 committed:
Resolved net::ERR_CONNECTION_REFUSED error in Dockerized environment by: (#17)

* WIP - add zustand

* Refactor authentication flow and API integration

- Introduced a new API utility for handling generic API calls.
- Updated authentication store to manage login, registration, and user details.
- Enhanced routing logic in App component to differentiate between public and protected routes.
- Added user details form validation schema.
- Updated Login and Register pages to utilize the new authentication store and handle navigation based on user authentication status.
- Adjusted import paths for consistency.
- Conditionally showing user details form.

* Fix import path for HomePage component in App.tsx

* Update import paths for consistency in Auth pages and enhance error handling in LoginPage

* Resolved net::ERR_CONNECTION_REFUSED error in Dockerized environment by:
- Configuring Vite proxy in vite.config.ts to forward '/api' requests to 'http://backend:3000', enabling frontend-backend communication within Docker network.
- Updated axiosConfig.ts baseURL from 'http://localhost:3000/api/v1' to '/api/v1' to use relative URLs, leveraging the Vite proxy for correct API routing.

* feat/dashboard: Updated home page, added dashboard components, and created style guide
- Updated home page to serve as main dashboard.
- Added sidebar for desktop navigation.
- Added mobile nav for bottom navigation.
- Created tabs with empty dashboard, transactions, budgets, and savings components.
- Added styleGuide.css in styles folder with design colors .

* feat(budget): Create AddBudgetButton
- Added button with dialog form to create budgets
- Integrated budget store for data management
- Used constants and types for form options and validation

* .

* Fixed edit form in BudgetDialog.tsx to pre-fill category field by mapping capitalized budget.category to lowercase CATEGORY_OPTIONS value.

* Added logic to BudgetFormFields.tsx to prevent duplicate category and theme selections by disabling and reducing opacity of used options in dropdowns

* Added spending input

* deleted unsedfile

* Calculated totalMaximum, totalSpending, and totalRemaining in budget store.
Added pieChart and spendingSummary components.

* feat(nav): Add logout button to Sidebar and MobileNav with navigation to landing page

* chore: update bun.lock

* Refactor Auth flow and change eslint config

* Resolve tests

* Refactor Workflow

* turn off samesite for preview

---------

Co-authored-by: Run Qi (Jack) Li <li.jack0707@gmail.com>
Co-authored-by: affan <syedaffan880@gmail.com>
Co-authored-by: kayoMichael <michaelxkayo@hotmail.com>

Shaimaa01 committed:
Resolved net::ERR_CONNECTION_REFUSED error in Dockerized environment by: (#17)

* WIP - add zustand

* Refactor authentication flow and API integration

- Introduced a new API utility for handling generic API calls.
- Updated authentication store to manage login, registration, and user details.
- Enhanced routing logic in App component to differentiate between public and protected routes.
- Added user details form validation schema.
- Updated Login and Register pages to utilize the new authentication store and handle navigation based on user authentication status.
- Adjusted import paths for consistency.
- Conditionally showing user details form.

* Fix import path for HomePage component in App.tsx

* Update import paths for consistency in Auth pages and enhance error handling in LoginPage

* Resolved net::ERR_CONNECTION_REFUSED error in Dockerized environment by:
- Configuring Vite proxy in vite.config.ts to forward '/api' requests to 'http://backend:3000', enabling frontend-backend communication within Docker network.
- Updated axiosConfig.ts baseURL from 'http://localhost:3000/api/v1' to '/api/v1' to use relative URLs, leveraging the Vite proxy for correct API routing.

* feat/dashboard: Updated home page, added dashboard components, and created style guide
- Updated home page to serve as main dashboard.
- Added sidebar for desktop navigation.
- Added mobile nav for bottom navigation.
- Created tabs with empty dashboard, transactions, budgets, and savings components.
- Added styleGuide.css in styles folder with design colors .

* feat(budget): Create AddBudgetButton
- Added button with dialog form to create budgets
- Integrated budget store for data management
- Used constants and types for form options and validation

* .

* Fixed edit form in BudgetDialog.tsx to pre-fill category field by mapping capitalized budget.category to lowercase CATEGORY_OPTIONS value.

* Added logic to BudgetFormFields.tsx to prevent duplicate category and theme selections by disabling and reducing opacity of used options in dropdowns

* Added spending input

* deleted unsedfile

* Calculated totalMaximum, totalSpending, and totalRemaining in budget store.
Added pieChart and spendingSummary components.

* feat(nav): Add logout button to Sidebar and MobileNav with navigation to landing page

* chore: update bun.lock

* Refactor Auth flow and change eslint config

* Resolve tests

* Refactor Workflow

* turn off samesite for preview

---------

Co-authored-by: Run Qi (Jack) Li <li.jack0707@gmail.com>
Co-authored-by: affan <syedaffan880@gmail.com>
Co-authored-by: kayoMichael <michaelxkayo@hotmail.com>

affan880 committed:
Refactor transactions model in schema.prisma to remove unused fields and streamline structure

affan880 committed:
Remove bun.lock file and update server tests workflow to improve Bun installation caching and add retry logic for dependency installation.

kayoMichael commented:
Close PR
